src/math/crypto/curve/ec_point.zig

```diff
@@ -0,0 +1,149 @@
+// code ported from starknet-curve:
+// https://github.com/xJonathanLEI/starknet-rs/blob/0857bd6cd3bd34cbb06708f0a185757044171d8d/starknet-curve/src/ec_point.rs
+const Felt252 = @import("../../fields/starknet.zig").Felt252;
+const ALPHA = @import("./curve_params.zig").ALPHA;
+const BETA = @import("./curve_params.zig").BETA;
+
+pub const ProjectivePoint = struct {
+    const Self = @This();
+
+    x: Felt252,
+    y: Felt252,
+    z: Felt252,
+    infinity: bool,
+
+    pub fn fromAffinePoint(p: AffinePoint) Self {
+        return .{
+            .x = p.x,
+            .y = p.y,
+            .z = Felt252.one(),
+            .infinity = false,
```

Comment:
If you apply the previous default suggestions, `.z` and `.infinity` are not needed anymore.